packages/core/src/tenants/SystemContext.ts

```diff
@@ -11,25 +20,44 @@ const { table, fields } = convertToIdentifiers(Systems);
 export default class SystemContext {
   static shared = new SystemContext();
   public storageProviderConfig: StorageProviderData | undefined;
+  public hostnameProviderConfig: HostnameProviderData | undefined;
+
+  async loadProviderConfigs(pool: CommonQueryMethods) {
+    this.storageProviderConfig = await this.loadConfig<StorageProviderData>(
+      pool,
+      StorageProviderKey.StorageProvider,
+      storageProviderDataGuard
+    );
+    this.hostnameProviderConfig = await this.loadConfig<HostnameProviderData>(
+      pool,
+      CloudflareKey.HostnameProvider,
```

Comment:
Should this be `HostnameProviderKey.HostnameProvider` or something more generic?

Reply:
No, since the logic is all about Cloudfalre. If we have other service in the future, we can add a new key.